mono/mini/cfold.c

```diff
@@ -258,6 +258,14 @@ mono_constant_fold_ins (MonoCompile *cfg, MonoInst *ins, MonoInst *arg1, MonoIns
 			dest->inst_c0 = arg1->inst_c0;
 		}
 		break;
+	case OP_FMOVE:
+		if (arg1->opcode == OP_R8CONST || arg1->opcode == OP_R4CONST) {
+			ALLOC_DEST (cfg, dest, ins);
+			dest->opcode = arg1->opcode;
+			dest->sreg1 = -1;
+			dest->inst_p0 = arg1->inst_p0;
+		}
```

Comment:
This might lead to worse code because it needs to load the same r4/r8 immediate many times.

Reply:
Nm, it was already there.

Reply:
I just moved it next to the other OP_xMOVE cases and added same treatment as for OP_R8CONST.